[Serializer.encoder] Added yml in format

FroggDev opened the pull request:
Added yml format in valid list

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        | 

Added yml as valid format to be able to use converter with .yml file without addind a custom converter in services

example :
https://github.com/FroggDev/Symfony_phpunit_advanced/blob/master/src/Serializer/YamlEncoder.php
https://github.com/FroggDev/Symfony_phpunit_advanced/blob/master/src/Exporter/ExportContact.php

an export of contact.yml require a custom YamlEncoder added in services.yaml
(another solution is to add a test if ext=yml then ext=yaml but it is really not a clean solution)

FroggFr committed:
[Serializer.encoder] Added yml in format

Added yml format in valid list

fabpot commented on src/Symfony/Component/Serializer/Encoder/YamlEncoder.php:
```diff
@@ -54,7 +54,7 @@ public function encode($data, $format, array $context = array())
     public function supportsEncoding($format)
     {
-        return self::FORMAT === $format;
+        return in_array($format, self::FORMAT);
```
I would add `, true` at the end of `in_array` (same below)

xabbuh commented on src/Symfony/Component/Serializer/Encoder/YamlEncoder.php:
```diff
@@ -22,7 +22,7 @@
 class YamlEncoder implements EncoderInterface, DecoderInterface
 {
-    const FORMAT = 'yaml';
+    const FORMAT = array('yaml', 'yml');
```
This constant may be used where changing the value from a scalar to an array could break existing code.

FroggDev replied:
I think you mean somewhere instead of somewhere.  
So, if  i let this CONST as it is and add another variable (named like $formatExt) equal to the array yaml & yml ? Would it be more reliable ?

nicolas-grekas replied:
I'd suggest to mark the const as deprecated, and add a new private one (eg `private const FORMATS = array(...)`

ogizanagi commented:
Thanks @FroggDev for your contribution. But honestly I'm not sure about this one. 
`$format` on the Serializer component point of view is only the canonical name of the input/output type. It has nothing to do with file extensions.

Also, the CsvEncoder is also able to handle tabulation-separated values or any other separator. That doesn't mean we should add `tsv` as a constant in `CsvEncoder`.

FroggFr committed:
FORMAT Marked as deprecated 

- added FORMAT as deprecated
- added FORMATS with all YAML Formats
- added strtolower to manage case verification
- added stric verification to in_array

FroggDev commented:
I updated the code to fit suggestions

fabpot commented:
@ogizanagi Shall we close?

ogizanagi commented:
I think so.
Anyway, thanks @FroggDev for this suggestion. I hope you understand the reasons for which I'm closing this.

FroggDev commented:
don't worry, i understand that i still not have enought knowledge on Symfony, but i will still try to make PR anyway if i think it could help ^_^
